### PR TITLE
[fix #470] Add tagged blog post to home page.

### LINKED ISSUE #470: Homepage blog section: add functionality to feature blog posts in favor of chronological order
Add functionality allowing the ability to tag blogs for priority placement in the blog content section on the homepage.

***Success Criteria***
- [ ] 3 column layout for (3) blog content items in that section currently on the homepage
- [ ] Leftmost column will feature tagged item, when a blog is tagged
- [ ] Middle and rightmost columns will feature most recent and, 2nd most recent chronologically posted blogs, when a blog is tagged
- [ ] Content will feature, left to right, most recent posts when tagged item is not present (not tag applied)

### PR DESCRIPTION
**Pull request by jpetto**
home page should now display 3 blog posts:

- if a post is found with the tag "story", that post should be shown last/on the right. the other two slots should show the two most recent posts (excluding the tagged post).
- if no post is found with the tag "story", the three most recent posts should be displayed

also adds a check for a missing blog image and provides a fallback, and adds some tests.

### CODE REVIEW

File: careers/careers/templates/careers/home.jinja [hunk 1]
```diff
@@ -133,14 +133,18 @@
...
+  <section class="mzp-l-card-third">
+  {% for post in recent_posts %}
     <div class="mzp-c-card mzp-c-card-medium mzp-has-aspect-16-9">
       <a href="{{ post.link }}" class="mzp-c-card-block-link">
         <div class="mzp-c-card-media-wrapper">
           <div class="lazy-image-container">
+          {% if post.image %}
             <img src="{{ post.image }}" alt="" class="mzp-c-card-image" />
+          {% else %}
+            <img src="{{ static('img/blog-fallback.png') }}" class="mzp-c-card-image" />
```
Comment (glogiotatidis) [inline careers/careers/templates/careers/home.jinja]: I guess we could use `|default` but maybe it's less readable. Up to you

File: careers/careers/wordpress.py [hunk 2]
```diff
@@ -98,3 +97,70 @@ def process_excerpt(excerpt):
...
+        del posts_data[featured_post_index]
+
+    return posts_data, featured_post_data
+
+
+def prepare_post_data(post_data):
+    # sometimes a post doesn't have a `post-large` image
+    try:
+        image = post_data['featured_media']['media_details']['sizes']['post-large']['source_url']  # noqa
+    except Exception:
```
Comment (glogiotatidis) [inline careers/careers/wordpress.py]: I always try to keep `Exceptions` as narrow as possible. In this case `KeyError`

File: careers/careers/wordpress.py [hunk 3]
```diff
@@ -98,3 +97,70 @@ def process_excerpt(excerpt):
...
+
+
+def get_posts():
+    # blog content isn't critical to the page. if the wordpress API fails
+    # for some reason, we can just move on and wait for the next build
+    try:
+        # 100 posts should give us over a year of content in which to find
+        # a featured/tagged post
+        posts_data = get_posts_data(num_posts=100)
+    except Exception:
```
Comment (glogiotatidis) [inline careers/careers/wordpress.py]: But in this case I remember it's complicated, so it's OK

File: careers/careers/templates/careers/home.jinja (hunk 1 above)
Comment (jpetto) [reply careers/careers/templates/careers/home.jinja]: hm, no, i like your suggestion better. will update. 😄

**Pull request merged**